linkerd/app/gateway/src/lib.rs

```diff
@@ -122,7 +126,7 @@ where
             let allow = allow_discovery.clone();
             move |addr: NameAddr| {
                 if allow.matches(addr.name()) {
-                    Ok(addr)
+                    Ok(profiles::LogicalAddr(addr.into()))
```

Comment:
TIOLI: seems like there could be a `From` impl for `LogicalAddr`? nbd.

Reply:
I want this to go away in favor of a Param, anyway.